src/server/scripts/Commands/cs_premium.cpp

```diff
@@ -0,0 +1,221 @@
+
+class premium_commandscript : public CommandScript
+{
+public:
+    premium_commandscript() : CommandScript("premium_commandscript") { }
+
+    std::vector<ChatCommand> GetCommands() const override
+    {
+        static std::vector<ChatCommand> characterCommandTable
+        {
+            { "create",     SEC_GAMEMASTER, false, &HandlePremiumCharacterCreateCommand, "" },
+            { "delete",     SEC_GAMEMASTER, false, &HandlePremiumCharacterDeleteCommand, "" },
+            { "info",       SEC_GAMEMASTER, false, &HandlePremiumCharacterInfoCommand, "" }
+        };
+
+        static std::vector<ChatCommand> accountCommandTable
+        {
+            { "create",     SEC_GAMEMASTER, false, &HandlePremiumAccountCreateCommand, "" },
+            { "delete",     SEC_GAMEMASTER, false, &HandlePremiumAccountDeleteCommand, "" },
+            { "info",       SEC_GAMEMASTER, false, &HandlePremiumAccountInfoCommand, "" }
+        };
+        static std::vector<ChatCommand> commandTable =
+        {
+            { "account",    SEC_GAMEMASTER, false, nullptr, "", accountCommandTable},
+            { "character",  SEC_GAMEMASTER, false, nullptr, "", characterCommandTable}
+        };
+        static std::vector<ChatCommand> premiumCommandTable =
+        {
+            { "premium",    SEC_GAMEMASTER, false, nullptr, "", commandTable}
+        };
+        return premiumCommandTable;
+    }
+
+    static bool HandlePremiumCharacterInfoCommand(ChatHandler* handler, const char* /*args*/)
+    {
+        Player* target = handler->getSelectedPlayer();
+        string playerName = target->GetName().c_str();
+        if (!target)
+        {
+            (ChatHandler(handler->GetSession())).PSendSysMessage(LANG_NO_CHAR_SELECTED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (handler->HasLowerSecurity(target, 0))
+            return false;
+
+        int premium_level = target->GetCharacterPremiumLevel(target->GetGUIDLow());
+        if (!premium_level)
+        {
+            (ChatHandler(handler->GetSession())).PSendSysMessage("No premium level available for character.");
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+        else
+            (ChatHandler(handler->GetSession())).PSendSysMessage("Character premium level: %u.", premium_level);
+
+        return true;
+    }
+
+    static bool HandlePremiumCharacterCreateCommand(ChatHandler* handler, const char* args)
+    {
+        if (!*args)
+            return false;
+
+        // Which premium level to be added
+        int premiumLevel = atoi((char*)args);
+
+        Player* target = handler->getSelectedPlayer();
+        string playerName = target->GetName().c_str();
+        if (!target)
+        {
+            handler->SendSysMessage(LANG_NO_CHAR_SELECTED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (handler->HasLowerSecurity(target, 0))
```

Comment:
Are those calls even needed? The commands are restricted anyways, right?

Reply:
I think so, commands are restricted by these checks only no?